Source/WTF/wtf/RefTrackerMixin.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "Compiler.h"
+#include "DataLog.h"
+#include "HashMap.h"
+#include "Lock.h"
+#include "StackShot.h"
+#include "StackTrace.h"
+
+#include <atomic>
+
+namespace WTF {
+
+// This file contains some tools for tracking references.
+// See Strong.h for an example of how to use it.
+
+#if ENABLE(REFTRACKER)
+template<typename T>
+struct RefTrackerLoggingDisabledScope {
+    WTF_MAKE_NONCOPYABLE(RefTrackerLoggingDisabledScope);
+    WTF_EXPORT_PRIVATE RefTrackerLoggingDisabledScope();
+    WTF_EXPORT_PRIVATE ~RefTrackerLoggingDisabledScope();
+};
+
+struct RefTracker {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    RefTracker() = default;
+    ~RefTracker() = default;
+
+    // NEVER_INLINE to make skipping frames more predictable.
+    WTF_EXPORT_PRIVATE void reportLive(void*);
+    WTF_EXPORT_PRIVATE void reportDead(void*);
+    WTF_EXPORT_PRIVATE void logAllLiveReferences();
+
+    Lock lock { };
```

Comment:
Could you annotate which data structure is guarded by the lock with `WTF_GUARDED_BY_LOCK` for the static analyzer, please?

Reply:
I mean yeah it's obvious which one it is but it would be nice to get the compilation failures for the lack of holding the lock when using the map.